core/controllers/profile.py

```diff
@@ -370,31 +370,40 @@ def put(self):
 
 
 class UserInfoHandler(base.BaseHandler):
-    """Provides info about user."""
+    """Provides info about user. If user is not logged in,
+    return dict containing false as logged in status.
+    """
 
     GET_HANDLER_ERROR_RETURN_TYPE = feconf.HANDLER_TYPE_JSON
 
-    @acl_decorators.require_user_id
+    @acl_decorators.open_access
     def get(self):
         """Handles GET requests."""
-
-        user_actions = user_services.UserActionsInfo(self.user_id).actions
-        user_settings = user_services.get_user_settings(
-            self.user_id, strict=False)
-        self.render_json({
-            'is_moderator': (
-                user_services.is_at_least_moderator(self.user_id)),
-            'is_admin': user_services.is_admin(self.user_id),
-            'is_super_admin': (
-                current_user_services.is_current_user_super_admin()),
-            'is_topic_manager': user_services.is_topic_manager(self.user_id),
-            'can_create_collections': bool(
-                role_services.ACTION_CREATE_COLLECTION in user_actions),
-            'preferred_site_language_code': (
-                user_settings.preferred_site_language_code),
-            'username': user_settings.username,
-            'user_is_logged_in': True
-        })
+        if self.user_id:
+            user_actions = user_services.UserActionsInfo(self.user_id).actions
+            user_settings = user_services.get_user_settings(
+                self.user_id, strict=False)
+
+        if self.user_id and user_settings.username:
```

Comment:
We already have self.username in base.py, should we use that instead? I think the base.py properties are used as the "source of truth" for the checks.